### PR TITLE
ref(sudoModal): Convert SudoModal to Typescript

### PR DESCRIPTION
**Pull request by priscilawebdev**
Convert SudoModal to Typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/sudoModal.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import TextBlock from 'app/views/settings/components/text/textBlock';
+import U2fContainer from 'app/components/u2f/u2fContainer';
+import space from 'app/styles/space';
+import {Client} from 'app/api';
+
+type Props = WithRouterProps & {
+  api: Client;
+  closeModal: () => void;
+  Header: typeof ModalHeader;
+  Body: typeof ModalBody;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/modals/sudoModal.tsx:24]: I think there is a `ModalProps` (or something like that) type for these

File: src/sentry/static/sentry/app/components/modals/sudoModal.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+   * expects a function that returns a Promise
+   */
+  retryRequest?: () => Promise<any>;
+};
+
+type State = {
+  error: boolean;
+  busy: boolean;
+};
+class SudoModal extends React.Component<Props, State> {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/modals/sudoModal.tsx:39]: nit: Line break before class declaration (maybe we configure https://eslint.org/docs/rules/padding-line-between-statements for it)

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/modals/sudoModal.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/modals/sudoModal.tsx:39]: sometimes I cann't see this line breaks in my code editor, I will have to figure out what is going on. I'll fix it...thanks :)

File: src/sentry/static/sentry/app/components/modals/sudoModal.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/modals/sudoModal.tsx:24]: oh yeah...the ModalRenderProps...lemme update it...thanks!

**Pull request merged**